src/ui/webeiddialog.cpp

```diff
@@ -356,12 +356,16 @@ void WebEidDialog::connectOkToCachePinAndEmitSelectedCertificate(
 
 void WebEidDialog::onRetryImpl(const QString& error)
 {
-    showPage(Page::WAITING);
-
-    const auto result =
-        QMessageBox::warning(this, tr("Retry?"), tr("Error occurred: ") + error,
-                             QMessageBox::StandardButtons(QMessageBox::Yes | QMessageBox::No));
-    emit result == QMessageBox::Yes ? retry() : reject();
+    ui->connectCardLabel->setText(error);
+    ui->selectCardPageTitleLabel->setText(tr("Error occurred"));
+    ui->cardChipIcon->setPixmap(QStringLiteral(":/images/id-card.svg"));
+    showPage(Page::INSERT_CARD);
```

Comment:
I initially thought to create a separate page for errors, but I guess reusing the *Insert card* page is better.

However, I would suggest renaming `Page::INSERT_CARD` to `Page::FAILURE` and renaming the labels accordingly (*failureLabel*, *failureTitleLabel* etc), otherwise there is semantic mismatch between variable names and actual usage.